src/utilities/common/canonicalStringify.ts

```diff
@@ -0,0 +1,78 @@
+// Like JSON.stringify, but with object keys always sorted in the same order.
```

Comment:
```suggestion
/**
 * Like JSON.stringify, but with object keys always sorted in the same order.
 *
 * To achieve performant sorting, this function uses a trie to store the sorted
 * key names of objects that have already been encountered, bringing "memoized"
 * sort actions down ot O(n) from O(n log n).
 *
 * As a drawback, this function will add a little bit more memory for every object
 * is called with that has different (more, less, a different order) keys than
 * in the past.
 *
 * In a typical application, this should not play a significant role, as
 * `canonicalStringify` will be called for only a limited number of object shapes,
 * and the cache will not grow beyond a certain point.
 * But in some edge cases, this could be a problem, so we provide a `reset` method
 * that will clear the cache and could be called at intervals from user.
 * */
```

I'd add this, to also give a bit of guidance on users what memory growth they can expect here (and that it will likely "flat out" at some point).